taskchampion/src/server/cloud/aws.rs

```diff
@@ -0,0 +1,628 @@
+    }
+
+    #[test]
+    fn list() {
+        let Some((mut svc, pfx)) = make_service() else {
+            return;
+        };
+        let mut names: Vec<_> = (0..20).map(|i| pfx(&format!("pp-{i:02}"))).collect();
+        names.sort();
+        // Create 20 objects that will be listed.
+        for n in &names {
+            svc.put(n, b"data").unwrap();
+        }
+        // And another object that should not be included in the list.
+        svc.put(&pfx("xxx"), b"data").unwrap();
+
+        let got_objects: Vec<_> = svc.list(&pfx("pp-")).collect::<Result<_>>().unwrap();
+        let mut got_names: Vec<_> = got_objects.into_iter().map(|oi| oi.name).collect();
+        got_names.sort();
+        assert_eq!(
+            got_names
+                .iter()
+                .map(|b| String::from_utf8(b.to_vec()).unwrap())
+                .collect::<Vec<_>>(),
+            names
+                .iter()
+                .map(|b| String::from_utf8(b.to_vec()).unwrap())
+                .collect::<Vec<_>>()
+        );
+
+        // Clean up.
+        for n in got_names {
+            svc.del(&n).unwrap();
+        }
+        svc.del(&pfx("xxx")).unwrap();
+    }
+
+    #[test]
+    fn compare_and_swap_create() {
+        let Some((mut svc, pfx)) = make_service() else {
+            return;
+        };
+
+        assert!(svc
+            .compare_and_swap(&pfx("testy"), None, b"bar".to_vec())
+            .unwrap());
+        let got = svc.get(&pfx("testy")).unwrap();
+        assert_eq!(got, Some(b"bar".to_vec()));
+
+        // Clean up.
+        svc.del(&pfx("testy")).unwrap();
+    }
+
+    #[test]
+    fn compare_and_swap_matches() {
+        let Some((mut svc, pfx)) = make_service() else {
+            return;
+        };
+
+        // Create the existing file, with two generations.
+        svc.put(&pfx("testy"), b"foo1").unwrap();
+        svc.put(&pfx("testy"), b"foo2").unwrap();
+        assert!(svc
+            .compare_and_swap(&pfx("testy"), Some(b"foo2".to_vec()), b"bar".to_vec())
+            .unwrap());
+        let got = svc.get(&pfx("testy")).unwrap();
+        assert_eq!(got, Some(b"bar".to_vec()));
+
+        // Clean up.
+        svc.del(&pfx("testy")).unwrap();
+    }
+
+    #[test]
+    fn compare_and_swap_expected_no_file() {
+        let Some((mut svc, pfx)) = make_service() else {
+            return;
+        };
+
+        svc.put(&pfx("testy"), b"foo1").unwrap();
+        assert!(!svc
+            .compare_and_swap(&pfx("testy"), None, b"bar".to_vec())
+            .unwrap());
+        let got = svc.get(&pfx("testy")).unwrap();
+        assert_eq!(got, Some(b"foo1".to_vec()));
+
+        // Clean up.
+        svc.del(&pfx("testy")).unwrap();
+    }
+
+    #[test]
+    fn compare_and_swap_mismatch() {
+        let Some((mut svc, pfx)) = make_service() else {
+            return;
+        };
+
+        // Create the existing file, with two generations.
+        svc.put(&pfx("testy"), b"foo1").unwrap();
```

Comment:
Again, I do not understand the purpose of the original `put`. (Assuming we're only trying to test taskchampion here, and not the aws library or service.)